File: Arrays/Algorithms/Searching/Unsorted/sublist_subarray_search.py
def sublist_search(arr, sub):
	n, m = len(arr), len(sub)
	for i in range(n - m + 1):
		if arr[i:i + m] == sub:
			return i
	return -1

if __name__ == "__main__":
	n, m = map(int, input().split())
	arr = list(map(int, input().split()))
	sub = list(map(int, input().split()))
	print(sublist_search(arr, sub))

File: Arrays/Algorithms/Searching/Unsorted/sublist_subarray_search.cpp
#include <bits/stdc++.h>
using namespace std;

int sublistSearch(const vector<int>& arr, const vector<int>& sub) {
	int n = arr.size(), m = sub.size();
	for (int i = 0; i <= n - m; i++) {
		int j = 0;
		while (j < m && arr[i + j] == sub[j]) j++;
		if (j == m) return i;
	}
	return -1;
}

int main() {
	int n, m;
	cin >> n >> m;
	vector<int> arr(n), sub(m);
	for (int i = 0; i < n; i++) cin >> arr[i];
	for (int i = 0; i < m; i++) cin >> sub[i];
	cout << sublistSearch(arr, sub) << endl;
	return 0;
}
